File: ClampingDevice/DTOs/ClampingStatsDto.cs
namespace ClampingDevice.DTOs;

public class ClampingStatsDto
{
    public int TotalClampings { get; set; }
    public int TotalUnclampings { get; set; }
    public int FailedClampings { get; set; }
    public DateTime? RecentClampingTimestamp { get; set; }
    public DateTime? RecentUnclampingTimestamp { get; set; }
    public int DevicesInvolved { get; set; }
}

File: Ngclient/src/app/_models/clampingStatsDto.ts
export interface ClampingStatsDto {
  totalClampings: number;
  totalUnclampings: number;
  recentClampingTimestamp: Date | null;
  recentUnclampingTimestamp: Date | null;
  failedClampings: number;
  devicesInvolved: number;
}
